DependencyInjection/Configuration.php

```diff
@@ -380,30 +394,47 @@ private function configureDbalDriverNode(ArrayNodeDefinition $node): void
      */
     private function addOrmSection(ArrayNodeDefinition $node): void
     {
+        // Key that should not be rewritten to the entity-manager config
+        $excludedKeys = [
+            'default_entity_manager' => true,
+            'auto_generate_proxy_classes' => true,
+            'enable_lazy_ghost_objects' => true,
+            'proxy_dir' => true,
+            'proxy_namespace' => true,
+            'resolve_target_entities' => true,
+            'resolve_target_entity' => true,
+            'controller_resolver' => true,
+        ];
+
         $node
             ->children()
                 ->arrayNode('orm')
                     ->beforeNormalization()
-                        ->ifTrue(static function ($v) {
+                        ->ifTrue(static function ($v) use ($excludedKeys) {
                             if (! empty($v) && ! class_exists(EntityManager::class)) {
                                 throw new LogicException('The doctrine/orm package is required when the doctrine.orm config is set.');
                             }
 
-                            return $v === null || (is_array($v) && ! array_key_exists('entity_managers', $v) && ! array_key_exists('entity_manager', $v));
+                            if ($v === null) {
+                                return false;
```

Comment:
Yep, I reversed it because a null value here was interesting to get into the normalization callback that would create the empty default entity manager configuration which we are now doing in the Extension.
This what not covered before but I added a test case.
And because now we are now sure to enter the normalization callback with an array I removed cast to an array there